WordPress/Classes/ViewRelated/Post/AbstractPostListViewController.swift

```diff
@@ -0,0 +1,1054 @@
+        
+        return syncHelper
+    }()
+    
+    var noResultsView : WPNoResultsView?
+    var postListFooterView : PostListFooterView!
+    var animatedBox : WPAnimatedBox?
+    
+    @IBOutlet var filterButton : NavBarTitleDropdownButton!
+    @IBOutlet var rightBarButtonView : UIView!
+    @IBOutlet var searchButton : UIButton!
+    @IBOutlet var addButton : UIButton!
+    @IBOutlet var searchWrapperView : UIView! // Used on iPhone for presenting the search bar.
+    @IBOutlet var authorsFilterView : UIView! // Search lives here on iPad
+    @IBOutlet var authorsFilterViewHeightConstraint : NSLayoutConstraint!
+    @IBOutlet var searchWrapperViewHeightConstraint : NSLayoutConstraint!
+    
+    var searchController : WPSearchController! // Stand-in for UISearchController
+    private var allPostListFilters = [String:[PostListFilter]]()
+    var recentlyTrashedPostObjectIDs = [NSManagedObjectID]() // IDs of trashed posts. Cleared on refresh or when filter changes.
+
+    private var needsRefreshCachedCellHeightsBeforeLayout = false
+    
+    // MARK: - Lifecycle
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        tableView = postListViewController.tableView
+        refreshControl = postListViewController.refreshControl
+        
+        refreshControl?.addTarget(self, action: #selector(AbstractPostListViewController.refresh(_:)), forControlEvents: .ValueChanged)
+        
+        configureCellsForLayout()
+        configureTableView()
+        configureFooterView()
+        configureNavbar()
+        configureSearchController()
+        configureAuthorFilter()
+        
+        WPStyleGuide.configureColorsForView(view, andTableView: tableView)
+        tableView.reloadData()
+    }
+    
+    override func viewWillAppear(animated: Bool) {
+        super.viewWillAppear(animated)
+        configureNoResultsView()
+    }
+    
+    override func viewDidAppear(animated: Bool) {
+        super.viewDidAppear(animated)
+        
+        automaticallySyncIfAppropriate()
+        NSNotificationCenter.defaultCenter().addObserver(self, selector: #selector(AbstractPostListViewController.handleApplicationDidBecomeActive(_:)), name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
+    
+    override func viewWillDisappear(animated: Bool) {
+        super.viewWillDisappear(animated)
+        
+        searchController.active = false
+        
+        NSNotificationCenter.defaultCenter().removeObserver(self, name: UIApplicationDidBecomeActiveNotification, object: nil)
+    }
+    
+    override func viewWillTransitionToSize(size: CGSize, withTransitionCoordinator coordinator: UIViewControllerTransitionCoordinator) {
+        super.viewWillTransitionToSize(size, withTransitionCoordinator: coordinator)
+        
+        coordinator.animateAlongsideTransition({[weak self] (context: UIViewControllerTransitionCoordinatorContext) -> () in
+            guard let strongSelf = self else {
```

Comment:
Pretty sure it's okay to just retain `self` here, since this method will be called whenever the (current + visible) view gets resized